crates/pallet-subspace/src/lib.rs

```diff
@@ -967,6 +972,18 @@ impl<T: Config> Pallet<T> {
     pub fn is_in_block_list(farmer_public_key: &FarmerPublicKey) -> bool {
         BlockList::<T>::contains_key(farmer_public_key)
     }
+
+    /// Size of the archived history of the blockchain in bytes
+    pub fn archived_history_size() -> u64 {
+        let archived_segments = RecordsRoot::<T>::count();
+        // `*2` because we need to include both data and parity pieces
+        let archived_segment_size = T::RecordedHistorySegmentSize::get() / T::RecordSize::get()
+            * u32::try_from(PIECE_SIZE)
+                .expect("Piece size is definitely small enough to fit into u32; qed")
```

Comment:
This produces the same result but simpler IMO.
```suggestion
            * PIECE_SIZE as u32
```

Reply:
It is simple and produces the same result until it is not and starts to silently drop data and cause unexpected behavior.

It is a good practice to use `uXX::from()` and `uXX::try_from()` to make error explicit if it happens. I didn't use it everywhere consistently yet (primarily not with `usize`), but I'm trying. That is an intentional design decision.

Reply:
That's fair, but now I'm kinda surprised that you don't think using `expect` always over `unwrap` is a good practice :(.

Reply:
I do think that is a good practice, it is just that in some cases `.unwrap()` is obvious and shorter. In blockchain code I have nothing against making it a hard rule to use `.expect()` in all cases.